### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by arnisimor**

### CODE REVIEW

File: kokibe.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ */
+/**
+ *
+ * @author 56130
+ */
+public class User {
+
+    private  int id ;
+    private String firstName;
+     private String lastName;
```
Comment (MingaudasVagonis) [inline kokibe.java:17]: Indentation

File: kokibe.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     private int age;
```
Comment (MingaudasVagonis) [inline kokibe.java:18]: Unused var, should remove

File: kokibe.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+/**
+ *
+ * @author 56130
+ */
+public class User {
+
+    private  int id ;
```
Comment (MingaudasVagonis) [inline kokibe.java:15]: Unused var, should remove.

File: kokibe.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    private String firstName;
+     private String lastName;
+     private int age;
+    private String profession;
+    private ArrayList<User> children;
```
Comment (MingaudasVagonis) [inline kokibe.java:20]: Unused var, should remove.

File: kokibe.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    public User(int id, String firstName, String lastName, int age) {
+
+        this.id = id;
```
Comment (MingaudasVagonis) [inline kokibe.java:24]: Remove since id unused

File: kokibe.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
+        this.profession = profession;
```
Comment (MingaudasVagonis) [inline kokibe.java:28]: Self assing, remove line.

File: kokibe.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+    public String getChildName(User child) {
+        if (!this.children.contains(child)) {
+            new Exception("Invalid argument!");
+        } else {
+            String name = null;
+            if (child.getFirstName() != null) {
+                name = child.getFirstName();
+            }
+            if (name == "Harry") {
```
Comment (MingaudasVagonis) [inline kokibe.java:40]: use equals instead of ==

File: kokibe.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                name.replace('r', 'p');
+            }
+            if (name != null || name.length() > 0) {
```
Comment (MingaudasVagonis) [inline kokibe.java:43]: refactor to one block and use apply results:
`if(name != null && name.length() > 0){
   if(name.equals("Harry")){
    name = name.replace('r', 'p');
   }
   this.name = name.concat(child.getLastName())
}`

File: kokibe.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+                name.concat(child.getLastName());
+            }
+        }
+        return this.getChildName(child);
```
Comment (MingaudasVagonis) [inline kokibe.java:47]: Remove line, infinite loop

**Review by MingaudasVagonis — CHANGES_REQUESTED** [review]

File: kokibe.java (hunk 1 above)
Comment (arnisimor) [reply kokibe.java:17]:
```suggestion
  private String lastName;
```

File: kokibe.java (hunk 2 above)
Comment (arnisimor) [reply kokibe.java:18]:
```suggestion
```

File: kokibe.java (hunk 4 above)
Comment (arnisimor) [reply kokibe.java:20]:
```suggestion
```

File: kokibe.java (hunk 5 above)
Comment (arnisimor) [reply kokibe.java:24]:
```suggestion

```

File: kokibe.java (hunk 6 above)
Comment (arnisimor) [reply kokibe.java:28]:
```suggestion

```

File: kokibe.java (hunk 7 above)
Comment (arnisimor) [reply kokibe.java:40]:
```suggestion
            if (name equals "Harry") {
```

File: kokibe.java (hunk 8 above)
Comment (arnisimor) [reply kokibe.java:43]:
```suggestion
if(name != null && name.length() > 0){ if(name.equals("Harry")){ name = name.replace('r', 'p'); } this.name = name.concat(child.getLastName()) }
```

File: kokibe.java (hunk 9 above)
Comment (arnisimor) [reply kokibe.java:47]:
```suggestion

```

**Pull request merged**